runelite-client/src/main/java/net/runelite/client/ui/ClientUI.java

```diff
@@ -198,6 +210,22 @@ public void onConfigChanged(ConfigChanged event)
 		});
 	}
 
+	@Subscribe
+	public void onGameStateChanged(GameStateChanged event)
+	{
+		justLoggedIn = event.getGameState() == GameState.LOGGED_IN;
```

Comment:
So apparently LOGGED_IN fires every time you move to a new region haha (where the map says loading and loads the new region). Not very well named at all. Pretty sure the people in the discord were saying we need to listen for certain packets that are only sent on login.

Reply:
Oof, can you remember who mentioned those packets? I'll try to prod them for more information.

Reply:
To test for logged in we generally test for login screen to logged in or world hop to logged in.

Reply:
Oh so if last state was LOGIN_SCREEN and the new state is LOGGED_IN? I like that idea!

Reply:
I would expect the username/player to be in place at the time of this event, though?

Reply:
I would just set it on CLICK_TO_PLAY screen, reset on LOGIN_SCREEN?

Reply:
I was just hoping we could find a way to avoid "updating" the title/user name every LOGGED_IN event. It works, but it just does more than it should and idk if there are perf implications of constantly changing the frame's title. Ideally there is an event or something that can tell us if a log in occurs, and that's the only time the event is fired.